Source/Particles/Collision/BinaryCollision/NuclearFusion/BoschHaleFusionCrossSection.H

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2022 Remi Lehe
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef DEUTERIUM_TRITIUM_FUSION_CROSS_SECTION_H
+#define DEUTERIUM_TRITIUM_FUSION_CROSS_SECTION_H
+
+#include "Particles/Collision/BinaryCollision/BinaryCollisionUtils.H"
+#include "Utils/WarpXConst.H"
+
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+/**
+ * \brief Computes the fusion cross section, using the analytical fits given in
+ * H.-S. Bosch and G.M. Hale 1992 Nucl. Fusion 32 611
+ *
+ * @param[in] E_kin_star the kinetic energy of the reactants in their center of mass frame, in SI units.
+ * @param[in] fusion_type indicates which fusion reaction to calculate the cross-section for
+ * @return The total cross section in SI units (square meters).
+ */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+amrex::ParticleReal BoschHaleFusionCrossSection (
+    const amrex::ParticleReal& E_kin_star,
+    const NuclearFusionType& fusion_type,
+    const amrex::ParticleReal& m1,
+    const amrex::ParticleReal& m2 )
+{
+    using namespace amrex::literals;
+
+    constexpr amrex::ParticleReal joule_to_keV = 1.e-3_prt/PhysConst::q_e;
+    const amrex::ParticleReal E_keV = E_kin_star*joule_to_keV;
+
+    // If kinetic energy is 0, return a 0 cross section and avoid later division by 0.
+    if (E_keV == 0._prt) {return 0._prt;}
+
+    // Compute the Gamow constant B_G (in keV^{1/2})
+    // (See Eq. 3 in H.-S. Bosch and G.M. Hale 1992 Nucl. Fusion 32 611)
+    const amrex::ParticleReal m_reduced = m1 / (1._prt + m1/m2);
+    amrex::ParticleReal B_G = MathConst::pi * PhysConst::alpha * 1._prt * 1._prt
```

Comment:
What is the reason for the two unity factors here: `... * 1._prt * 1._prt * ...`?

Reply:
This was just to emphasize the fact that this is the formula for `Z = 1`, but it looks a bit silly.
I will replace this by a proper comment.